src/Microsoft.OData.Core/Uri/NodeToStringBuilder.cs

```diff
@@ -470,7 +471,240 @@ internal String TranslateNode(QueryNode node)
+
+            BinaryOperatorKind binaryNodeKind = node.OperatorKind;
+
+            string binaryOperator = this.BinaryOperatorNodeToString(binaryNodeKind);
+            string[] binarySeparator = { binaryOperator };
+            string[] substrings = translatedNode.Trim().Split(binarySeparator, StringSplitOptions.RemoveEmptyEntries);
+            string leftBinary = substrings[0].Trim();
+            string rightBinary = substrings[1].Trim();
+
+            if (leftRangeVariableNode != null && IsDifferentSource(filterClauseRangeVariable, leftRangeVariableNode))
+            {
+                leftBinary = "$it/" + leftBinary;
+            }
+
+            if (rightRangeVariableNode != null && IsDifferentSource(filterClauseRangeVariable, rightRangeVariableNode))
+            {
+                rightBinary = "$it/" + rightBinary;
+            }
+
+            return leftBinary + ' ' + binaryOperator + ' ' + rightBinary;
+        }
+
+        /// <summary>
+        /// Bind $it to the <see cref="InNode"/> translated string.
+        /// </summary>
+        /// <param name="node">node to bind.</param>
+        /// <param name="filterClauseRangeVariable">The <see cref="FilterClause"/> range variable.</param>
+        /// <returns>The translated string with $it binding.</returns>
+        private string BindInNode(InNode node, ResourceRangeVariable filterClauseRangeVariable)
+        {
+            string translatedNode = this.TranslateNode(node);
+            ResourceRangeVariableReferenceNode leftRangeVariableNode = GetResourceRangeVariableReferenceNode(node.Left);
+
+            string inOperator = ExpressionConstants.KeywordIn;
+            string[] inSeparator = { inOperator };
+            string[] substrings = translatedNode.Trim().Split(inSeparator, StringSplitOptions.RemoveEmptyEntries);
+            string leftIn = substrings[0];
+            string rightIn = substrings[1];
+
+            if (leftRangeVariableNode != null && IsDifferentSource(filterClauseRangeVariable, leftRangeVariableNode))
+            {
+                leftIn = "$it/" + leftIn;
+                translatedNode = leftIn + inOperator + rightIn;
+            }
+            return translatedNode;
+        }
+
+        /// <summary>
+        /// Bind $it to the <see cref="AnyNode"/> translated string.
+        /// </summary>
+        /// <param name="node">node to bind.</param>
+        /// <param name="filterClauseRangeVariable">The <see cref="FilterClause"/> range variable.</param>
+        /// <returns>The translated string with $it binding.</returns>
+        private string BindAnyNode(AnyNode node, ResourceRangeVariable filterClauseRangeVariable)
+        {
+            string translatedNode = this.TranslateNode(node);
+            ResourceRangeVariableReferenceNode leftRangeVariableNode = GetResourceRangeVariableReferenceNode(node);
+
+            const string slashAny = "/any";

```

Comment:
Maybe have this as a `SlashAny` constant in `ExpressionConstants` class?

Reply:
We have `KeywordAny` in `ExpressionConstants` which represents `any`. I don't think we have more instances we will be using `/any`

Reply:
As above, if this was a performance critical hotspot with many iterations it would be worth defining SlashAny, but in this case it's probably not worth it.

---
In reply to: [574293978](https://github.com/OData/odata.net/pull/1991#discussion_r574293978) [](ancestors = 574293978)